lib/aurora_bootstrapper/exporter.rb

```diff
@@ -11,23 +11,32 @@ def initialize( client:, prefix: "", export_bucket:, blacklisted_tables: "", whi
       @whitelisted_tables = whitelisted_tables.split(",")
       @blacklisted_fields = blacklisted_fields.split(",")
       @client             = client
+      @notifier           = Notifier.new
     end
 
     def export!
       @client.query( "set sql_mode='NO_BACKSLASH_ESCAPES'" )
+
       database_names.all? do | database_name |
         begin
           @client.query( "use `#{database_name}`" )
           database = Database.new database_name: database_name,
                                          client: @client,
                              blacklisted_tables: @blacklisted_tables,
                              whitelisted_tables: @whitelisted_tables,
-                             blacklisted_fields: @blacklisted_fields
+                             blacklisted_fields: @blacklisted_fields,
+                             export_date: @notifier.export_date
```

Comment:
How does this handle the use case when the user doesn't want a date in the path?

Reply:
why we do not want one? if there were no datetime subfolder, the data files would end up under e.g. backfill/
app1d/app1d-mysql1/

Reply:
If two db dumps are on going it could overwrite each other, leaving the data files in inconsistent mode

Reply:
Correct. While this is not the usecase Data Plat wants, this is how BI expects data to be exported.

Reply:
Then BI team needs to change their script logic. We cannot do if elif elif ... for different teams here. We define the folder structure.

Reply:
If BI team does not want it it can explicitly set EXPORT_DATE to be empty.

Reply:
This introduces a breaking behavior change: previously omitting `EXPORT_DATE` would have skipped it, but now the user will have to add the date, but make it blank. Is this what you would expect as an end user?

Setting `EXPORT_DATE` to empty will evaluate to empty string.
```
> EXPORT_DATE= irb
2.5.0 :001 > ENV['EXPORT_DATE']
 => "" 
```
Will the code gracefully handle this case?

Reply:
Add another env variable EXPORT_DATE_TURN_OFF

Reply:
Can you help me understand how this is not a breaking change?

Reply:
Synced up offline.